### DISCUSSION TITLE
Support for managing projects

### DISCUSSION THREAD

**Discussion by tomkerkhove** [community/community#3023] [Mobile] [answered]
Are there plans to support managing projects?

**Comment by candyho** [comment]
Hey thanks for your feedback! Feel free to upvote on this topic related to Projects to help boost awareness of this request! - https://github.com/github/feedback/discussions/3023 
Also appreciate if you can share more details with us in the thread above 👆 - how would you expect to manage projects on mobile? Do you expect to view the full kanban board on mobile? What are the key project features that you think we should cater on mobile?

**Reply by tomkerkhove** [reply]
I think the initial version can be as simple as just being able to view the project and moving issues to/from/between columns and reshuffling the order.

**Reply by jimhill** [reply]
This would be a fantastic addition. I really like the ability to use the mobile app for project management and prioritising work using the Kanban view would be exceptional. 

We spend a lot of time in the website Kanban view and this is exactly the sort of thing I feel that would personally maximise the utility of the app.

Initially it could be just the ability to create an entry/issue and drag between and inside columns.

**Reply by candyho** [reply]
Hey thanks @tomkerkhove and @jimhill for sharing your thoughts. Do you mind telling us in which context you think you'd probably want to access the project boards with our apps? For example, do you imagine yourself using projects in-app to get an overview of project status or update issue details on the board while you are cooking in the kitchen, or during break time in a car / on public transport etc.? Would love to understand more on specific user scenarios to see how we could shape this feature to fit the most of your needs. :)

**Reply by jimhill** [reply]
Hey @candyho - it's a little different as we are in lockdown here in the UK. But in general I use my iPad to get away from my development machine. So I switch context - great for more creative activities and thinking/planning time. Sometimes it's great in terms of planning in a small team, to brain dump issues in the same way I'd use todos applications. Scheduling and prioritise jobs for the week. General tidy up, organisation, delegation and planning activities. 
Often this will be in the evening at home away from a "work" environment. Or often work like this in cafes/bars/at the beach - the iPad's beauty for me is the context switching.
Hope that makes sense?

**Reply by pobengtsson** [reply]
Projects support would be fantastic. I really like the light weight approach taken i github projects as it is also versatile, eg I can have issues belong to multiple projects. I recognize my own needs a lot in @jimhill description.

**Reply by BOSALEEM** [reply]
> I think the initial version can be as simple as just being able to view the project and moving issues to/from/between columns and reshuffling the order.

**Reply by rafaelurben** [reply]
I think the most important features would be to create and edit notes and view all other cards. 👍 Of course, editing and moving all the other cards would also be very nice, but that's almost advanced - just opening the pull request on click would also be enough. (in my opinion)

EDIT: Why I think this is an important feature:
Probably everyone knows it - you get ideas everywhere and sometimes you just want to write them down somewhere. And I think a GitHub project is a very good place for that.

**Reply by BrentMifsud** [reply]
I agree with @tomkerkhove. Just being able to simply move issues to and from project, and between columns in the kanban board would be nice.

Even if we cant create projects and do things like change user permissions and stuff. I would be completely fine with just being able to move issues around, as that is my most common interaction with projects.

**Comment by cdce8p** [comment]
I would love if this could be added. Since, unfortunately the projects view doesn't even work in Safari (iPad OS), there is no option to view a project on mobile currently.

Thus for a start, I would be happy with just the ability to view them. Next on the list would be to be able to move issues between columns. From the broken experience in Safari I believe that touch might not be the best option to accomplish that task. It might be better to have a details view with a separate dropdown to select the column.

**Reply by kyleschmolze** [reply]
Yes! As someone who uses projects heavily it's a big bummer to have them completely missing from the app. Just even a view only version would help me. That, and the ability to see/edit project status of Issues while viewing an issue, would really cover my core needs.

**Reply by candyho** [reply]
Hey thanks @cdce8p and @kyleschmolze for sharing your thoughts. Do you mind telling us in which context you think you'd probably want to access the project boards with our apps? For example, do you imagine yourself using projects in-app to get an overview of project status or update issue details on the board while you are cooking in the kitchen, or during break time in a car / on public transport etc.? Would love to understand more on specific user scenarios to see how we could shape this feature to fit the most of your needs. :)

**Reply by kyleschmolze** [reply]
For me the most common use case is probably while I'm on a call but not by my computer. I need to be able to talk with my team about what we're working on, and what's up next in our todo list. For non-code projects I'll often have a Google Doc up on my phone for reference while taking a walk, or in the kitchen, just to be able to reference it and make good group decisions, but being free from the computer during the call :) So just the ability to do some basic organization would be all I'd need. Drag-and-drop to reorder, tag new issues into a project, and view the canban.

**Reply by kyleschmolze** [reply]
Just to follow up here, these screenshots from the website on mobile show really my main need. I receive notifications on my phone, and being able to triage issues into the proper column in the board is super helpful - otherwise I have to remember to do it when I get to my computer which makes the whole mobile app less useful. 

![image](https://user-images.githubusercontent.com/1200038/111036324-8d7c3e00-83d3-11eb-893b-3186d5a02ce1.png)

![image](https://user-images.githubusercontent.com/1200038/111036327-91a85b80-83d3-11eb-91bb-dadb3d719e0d.png)

**Comment by guylepage3** [comment]
Projects are a must for my teams to use the GitHub mobile app. I suggest the same functionality as the web app on an iPhone. Works very well.

**Comment by eliserichards** [comment]
When opening a project, I would like to see the first (leftmost) column of cards. This column would be scrollable up & down, and I would be able to swipe left/right to see other columns in the project.

**Comment by tvvocold** [comment]
hello guys, any update on this feature request?

**Reply by candyho** [reply]
Hey, adding the ability to move one issue from one project column to another is on our radar! Hopefully we have some updates to share with you all soon!

**Reply by guylepage3** [reply]
Amazing that that's happening @candyho. But  I don't think that's relevant to this issue... We are wanting "Projects" in the mobile app. It's a must for everyone.

**Comment by guylepage3** [comment]
@candyho, any update on this issue?

**Reply by rafaelurben** [reply]
https://github.com/github/feedback/discussions/3023#discussioncomment-1665645

**Reply by guylepage3** [reply]
@rafaelurben @candyho that is in regards to the mobile version of the web app. Not the iOS mobile app.

**Comment by candyho** [comment, marked as answer]
Hey👋just want to inform you we've just shipped [view and edit projects(beta) custom fields]( https://github.blog/changelog/2022-05-09-view-and-edit-projectsbeta-custom-fields-on-github-mobile/) to both iOS and Android last week! You can now edit custom text, number, date, and single-select fields for issues and pull requests that belong to the new projects experience from within the triage sheet. Try it out and let us know what you think! Stay tuned for more updates!

**Reply by archon810** [reply]
It works well, otherwise the Github app for Android would not be usable for us, so thank you for that.

Are there any updates on bringing full Projects support, for boards and list layouts, with some sort of editing capabilities?

**Reply by candyho** [reply]
Yes! We are actively working on improving Projects support on mobile, including the ability to easily edit custom fields on the go. Stay tuned for more updates soon!

**Reply by jimhill** [reply]
@candyho Great!

**Reply by Wladefant** [reply]
@tomkerkhove @candyho i think this should be marked as the answer

**Reply by Wladefant** [reply]
@amrehman @candyho this discussion is now resolved and can be closed

**Reply by Katli95** [reply]
> Yes! We are actively working on improving Projects support on mobile, including the ability to easily edit custom fields on the go. Stay tuned for more updates soon!

Where can one stay tuned? I'm very much looking forward to "draft item creation" support!

**Comment by Wladefant** [comment]
#35791
> <img width="1200" alt="188992932-ee7ee343-c1b8-46d7-b0c3-cf79337aecfb (1)" src="https://user-images.githubusercontent.com/7498102/195095187-72713259-a9f9-42cd-89e7-36d4dbfaeba3.png">
> 
> 👋🏼Hey everyone,
> 
> We're excited to introduce GitHub Projects support on GitHub Mobile, now in public beta! You can now view and manage projects on the go. We would love for you to try it out and give us your early feedback here -  Do you find the feature useful? What else do you want us to support and why?
> 
> ## How do I test this
> If you are already a beta user - simply update to the latest beta builds through TestFlight for iOS or Google Play Store for Android.
> 
> If you are not a beta user yet, join GitHub Mobile beta program through the links below:
> 
> [Join iOS TestFlight](https://testflight.apple.com/join/NLskzwi5)
> [Join Google Play Beta](https://play.google.com/apps/testing/com.github.android)
> 
> ## Feature highlights
> 
> - Access your projects through repo, organization, and user profiles
> - Search for your desired projects
> - View all of your project items 
> - View items as grouped and configured on web
> - All custom fields attached to an item are represented as metadata pill
> - Long press on an item to perform the following quick actions:
>   - Edit title
>   - Edit custom fields
>   - Close an issue
>   - Delete an item from project
>   - Share a project
> - Tap on the project header to switch to other project views
> 
> ## Known issues
> Large projects might take longer or fail to load, we are working on optimizing the performance of larger projects. Please stay tuned for more updates.
> 
> For more details please refer to this [blog post](https://github.blog/2022-10-11-on-the-go-with-github-projects-on-github-mobile-public-beta/).
> 
> Happy testing! ❤️

@tomkerkhove I think you should mark as the answer

**Reply by archon810** [reply]
Doesn't look like the board UI is supported yet, a pretty big omission.

**Reply by guylepage3** [reply]
> Doesn't look like the board UI is supported yet, a pretty big omission. 

Agreed.

The funny thing is that mobile web supports it. Go figure.. 🤪

**Comment by guylepage3** [comment]
Projects mobile does not work for Classic. Beta Projects STILL does not allow for notes.. A note should be able to line break. Even adding `<br />` tag does not line break. Until I can use notes I'll continue to use Classic.
